cdap-api/src/main/java/co/cask/cdap/api/mapreduce/MapReduceContext.java

```diff
@@ -193,4 +191,9 @@
    * @param resources Resources that each reducer should use.
    */
   void setReducerResources(Resources resources);
+
+  /**
+   * Return true if the execution was successful, false otherwise.
+   */
+  boolean isSuccessful();
```

Comment:
For the state, we probably need a state that means before running (in case something call it in `initialize`).